### PR TITLE
Minor refactoring for new provisioner

### PR DESCRIPTION
**Pull request by suquark**
This minor refactoring (moving a segment of code into a method) makes it easier and cleaner to integrate #1702. This trivial refactoring does not change any behavior of Skypilot.

Tested (run the relevant ones):

- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/backends/cloud_vm_ray_backend.py [hunk 1]
```diff
@@ -2256,57 +2256,66 @@ def _provision(self,
...
                         zone=stdout.strip())
+            self._finalize_provisioning_no_lock(handle, task,
+                                                prev_cluster_status, ip_list,
+                                                lock_path)
+            return handle
+
+    def _finalize_provisioning_no_lock(
+            self, handle: ResourceHandle, task: task_lib.Task,
+            prev_cluster_status: global_user_state.ClusterStatus,
+            ip_list: List[str], lock_path: str):
```
Comment (Michaelvll) [inline sky/backends/cloud_vm_ray_backend.py:2267]:
```suggestion
            ip_list: List[str], lock_path: str) -> None:
```

File: sky/backends/cloud_vm_ray_backend.py [hunk 2]
```diff
@@ -2256,57 +2256,66 @@ def _provision(self,
                     # initialization.
                     handle.launched_resources = handle.launched_resources.copy(
                         zone=stdout.strip())
+            self._finalize_provisioning_no_lock(handle, task,
+                                                prev_cluster_status, ip_list,
+                                                lock_path)
+            return handle
+
+    def _finalize_provisioning_no_lock(
```
Comment (Michaelvll) [inline sky/backends/cloud_vm_ray_backend.py:2264]: How about renaming it to `_update_after_cluster_provisioned()`?

**Review by Michaelvll — APPROVED** [review]
Thanks for submitting the PR @suquark! The refactoring looks good to me. Please run a simple `sky launch -c non-existing`, `sky launch -c existing` test before merging it. : )

**Comment by suquark** [comment]
test passed

**Pull request merged**